code/game/atoms_movable.dm

```diff
@@ -848,6 +848,9 @@
 /atom/movable/proc/do_attack_animation(atom/target, atom/tool)
 	set waitfor = 0
 
+	if(!tool)
```

Comment:
Alright, so should I add a warning should it ever enter that if statement?

First time i've heard of ASSERT so not sure of its usage.